metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1406,44 +1406,51 @@ class MetalsLanguageServer(
         .orNull
     }.asJava
 
-  private def supportedBuildTool(): Option[BuildTool] =
+  private def supportedBuildTool(): Future[Option[BuildTool]] =
     buildTools.loadSupported match {
```

Comment:
Maybe we should filter the ones with incompatible versions and show a warning for those?

Reply:
So one thing I thought about while doing this is in a situation where a user has an sbt and a mill file for example, I think it's important for the choice to take priority. Because let's say they are on an unsupported sbt version, but they want sbt, then it's probably better for them to choose sbt and then get a warning and "no build tool" message rather than have sbt filtered out, and mill chosen even if that's not what they want.

What do you think?

Reply:
Sure, that's probably better.